Configurations/unix-Makefile.tmpl

```diff
@@ -188,61 +188,111 @@ HTMLSUFFIX=html
-BIN_CXXFLAGS={- join(' ', '$(CXXFLAGS)', $target{bin_cxxflags} || ()) -}
-BIN_LDFLAGS={- join(' ', '$(LDFLAGS)', $target{bin_lflags} || ()) -}
+                                 (map { "-D".$_} @{$config{CPPDEFINES}}),
+                                 (map { "-I".$_} @{$config{CPPINCLUDES}}),
+                                 @{$config{CPPFLAGS}}) -}
+CFLAGS={- join(' ', @{$config{CFLAGS}}) -}
+CXXFLAGS={- join(' ', @{$config{CXXFLAGS}}) -}
+LDFLAGS= {- join(' ', @{$config{LDFLAGS}}) -}
+EX_LIBS= {- join(' ', @{$config{LDLIBS}}) -}
+
+MAKEDEPEND={- $config{makedepprog} -}
 
 PERL={- $config{perl} -}
 
-AR=$(CROSS_COMPILE){- $config{ar} -}
-ARFLAGS= {- join(' ', @{$config{arflags}}) -}
-RANLIB={- $config{ranlib} ? "\$(CROSS_COMPILE)$config{ranlib}" : "true"; -}
-RC= $(CROSS_COMPILE){- $target{rc} || "windres" -}
-RCFLAGS={- join(' ', @{$config{rcflags}}) -} {- $target{shared_rcflag} -}
+AR=$(CROSS_COMPILE){- $config{AR} -}
+ARFLAGS= {- join(' ', @{$config{ARFLAGS}}) -}
+RANLIB={- $config{RANLIB} ? "\$(CROSS_COMPILE)$config{RANLIB}" : "true"; -}
+RC= $(CROSS_COMPILE){- $config{RC} -}
+RCFLAGS={- join(' ', @{$config{RCFLAGS}}) -} {- $target{shared_rcflag} -}
+
+# We let the C compiler driver to take care of .s files. This is done in
+# order to be excused from maintaining a separate set of architecture
+# dependent assembler flags. E.g. if you throw -mcpu=ultrasparc at SPARC
+# gcc, then the driver will automatically translate it to -xarch=v8plus
+# and pass it down to assembler.
+AS={- $config{AS} ? "\$(CROSS_COMPILE)$config{AS}" : '$(CC) -c' -}
+ASFLAGS={- join(' ', @{$config{ASFLAGS}}) || '$(CFLAGS)' -}
+
 RM= rm -f
 RMDIR= rmdir
-TAR= {- $target{tar} || "tar" -}
-TARFLAGS= {- $target{tarflags} -}
-MAKEDEPEND={- $config{makedepprog} -}
+TAR= {- $target{TAR} || "tar" -}
+TARFLAGS= {- $target{TARFLAGS} -}
 
 BASENAME=       openssl
 NAME=           $(BASENAME)-$(VERSION)
 TARFILE=        ../$(NAME).tar
 
-# We let the C compiler driver to take care of .s files. This is done in
-# order to be excused from maintaining a separate set of architecture
-# dependent assembler flags. E.g. if you throw -mcpu=ultrasparc at SPARC
-# gcc, then the driver will automatically translate it to -xarch=v8plus
-# and pass it down to assembler.
-AS={- $config{as} ? "\$(CROSS_COMPILE)$config{as}" : '$(CC) -c' -}
-ASFLAGS={- join(' ', @{$config{asflags}}) || '$(CFLAGS)' -}
+##### Project flags ##################################################
+
+# Variables starting with CNF_ are common variables for all product types
+
+CNF_CPPFLAGS={- join(' ', $target{cppflags} || (),
+                          (map { "-D".$_} @{$config{defines}}),
+                          (map { "-I".$_} @{$config{includes}}),
+                          @{$config{cppflags}}) -}
+CNF_CFLAGS={- join(' ', $target{cflags} || (),
+                        @{$config{cflags}}) -}
+CNF_CXXFLAGS={- join(' ', $target{cxxflags} || (),
+                          @{$config{cxxflags}}) -}
```

Comment:
On related note, as problem ought to be somewhere else, when looking at linux-x86_64 file I couldn't help noticing that `CNF_CFLAGS=-pthread -m64`, while `CNF_CXXFLAGS=-std=c++11 -pthread`. I'd say that latter should also have -m64...

Reply:
Good point.  That's something to fix in `10-main.conf`

Reply:
There *is* another option, *IF* we can say for sure that any C++ flag will simply be an addition to existing C flags, then we can change the semantics of `cxxflags` to be *additional* to `cflags` instead of a replacement.  Your thoughts on this?

Note that it's safe to change the semantics for `cxxflags`, as it's new in 1.1.1-dev.

Reply:
> Your thoughts on this?

Well, *IF* we can say it for sure, then of course. Trouble is that we can't be 100% sure. It feels that it should work, as long as cxxflags is appended to cflags, but c++ driver doesn't actually have to recognize c-specific options. One example can be -std=c9x... Looking at 10-main modification before last one can start wondering wouldn't be nice if a flag could be *added* to both cflags and cxxflags with one line :-) BTW, default GNU make rules use $(TARGET_ARCH). This kind of gives you an idea that it might be appropriate to introduce similar notion [of common target-specific flag] even for us...